### PR TITLE
refactor resnet50 ttnn. GS + WH. Batch size 8,16,20 supported.

### PR DESCRIPTION
**Pull request by tt-nshanker**

### CODE REVIEW

File: ttnn/ttnn/operations/binary.py [hunk 1]
```diff
@@ -480,6 +480,7 @@ def add_and_apply_activation(
     activation: Optional[str] = None,
     memory_config: ttnn.MemoryConfig = ttnn.DRAM_MEMORY_CONFIG,
     dtype: Optional[ttnn.DataType] = None,
+    out_in_place: Optional[bool] = False,
```
Comment (cfjchu) [inline ttnn/ttnn/operations/binary.py:483]: Could we change this to `in_place` and add documentation for this function parameter?

**Review by eyonland — CHANGES_REQUESTED** [review]
Can we replace https://github.com/tenstorrent-metal/tt-metal/blob/main/models/experimental/resnet/tt/ttnn_functional_resnet.py with this version of resnet50 and update
https://github.com/tenstorrent-metal/tt-metal/blob/main/tests/ttnn/integration_tests/resnet/test_performance.py
?

**Comment by tt-nshanker** [comment]
> Can we replace https://github.com/tenstorrent-metal/tt-metal/blob/main/models/experimental/resnet/tt/ttnn_functional_resnet.py with this version of resnet50 and update https://github.com/tenstorrent-metal/tt-metal/blob/main/tests/ttnn/integration_tests/resnet/test_performance.py ?

I haven't added WH support yet. I will replace the old version with this once WH is ready too and also update test_performance.py

File: ttnn/ttnn/operations/binary.py [hunk 2]
```diff
@@ -486,6 +486,7 @@ def add_and_apply_activation(
     activation: Optional[str] = None,
     memory_config: ttnn.MemoryConfig = ttnn.DRAM_MEMORY_CONFIG,
     dtype: Optional[ttnn.DataType] = None,
+    in_place: Optional[bool] = False,
```
Comment (arakhmati) [inline ttnn/ttnn/operations/binary.py:489]: We already have an in-place version. With the underscore at the back.use that and don't modify this

**Review by arakhmati — CHANGES_REQUESTED** [review]

**Review by cfjchu — APPROVED** [review]

**Review by arakhmati — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Pull request merged**